6937: Replace Graphviz with Lineage Graph Component

labkey-nicka opened the pull request:
Closing and re-issuing PR. See #1031.

labkey-nicka committed:
ExperimentRunGraphView to JSP

labkey-nicka committed:
ExperimentRunGraphView to JSP

labkey-nicka committed:
Support async/await

labkey-nicka committed:
Support async/await

labkey-nicka committed:
Remove redundant webpack plugins

labkey-nicka committed:
Remove redundant webpack plugins

labkey-nicka committed:
Run Graph: initial app infrastructure

labkey-nicka committed:
Run Graph: initial app infrastructure

labkey-nicka committed:
Incorporate component styling

labkey-nicka committed:
Incorporate component styling

labkey-nicka committed:
Enable URL resolver

labkey-nicka committed:
Enable URL resolver

labkey-nicka committed:
Update Experiment.js fetch actions to return XMLHttpRequest
- see https://github.com/LabKey/labkey-api-js/pull/44

labkey-nicka committed:
Update Experiment.js fetch actions to return XMLHttpRequest
- see https://github.com/LabKey/labkey-api-js/pull/44

labkey-nicka committed:
Remove comment

labkey-nicka committed:
Remove comment

labkey-nicka committed:
Note about async/await

labkey-nicka committed:
Note about async/await

labkey-nicka committed:
Skip view generation for app

labkey-nicka committed:
Skip view generation for app

cnathe commented on experiment/package.json:
```diff
@@ -29,7 +29,7 @@
   },
   "dependencies": {
-    "@labkey/components": "0.41.0"
+    "@labkey/components": "0.42.0-fb-rungraph-lineage.4"
```
I haven't looked at your changes in the labkey-ui-component side for this package version yet, but do you think there is any need to bump the @labkey/components version in the other platform package.json files to keep them in sync? or is this set of change very specific to only the usage in experiment?

labkey-nicka replied:
Good idea. I've gone ahead and bumped all usages of `@labkey/components` in platform.

cnathe commented on experiment/src/client/RunGraph/util.ts:
```diff
@@ -0,0 +1,73 @@
+    target: string
+}
+
+export interface InitAppEventDetail<T> {
```
The plan is to fine somewhere more central to share the code below this line related to the registerApp stuff, right? Have you thought more about where that might go?

labkey-nicka replied:
We don't really have a good home for it at the moment. I think `@labkey/api` could suffice as long as the implementation remains agnostic of React, etc. `@labkey/components` could work but it feels more odd as it is not a component and more of a shim/util to help load apps in LKS.

If we port the implementation as-is we'll need a way to ensure the page registers the event listener before pages request to load the app.

cnathe reviewed (commented):
The platform webpack config changes look good and the RunGraph React implementation looks good with the registerApp option (I've added this as a Frontend Dev meeting topic to review / discuss / make the group aware). 

I'll let Kevin do the final review / approval of the Java and jsp changes.

labkey-nicka committed:
Navigate to lineage if available

labkey-nicka committed:
Navigate to lineage if available

labkey-nicka committed:
Bump @labkey/components for all modules

labkey-nicka committed:
Bump @labkey/components for all modules

labkey-kevink commented on .gitattributes:
```diff
@@ -2298,6 +2298,7 @@ experiment/src/org/labkey/experiment/controllers/exp/ExperimentRunGraphView.java
 experiment/src/org/labkey/experiment/controllers/exp/ExperimentRunGroupsView.java -text
 experiment/src/org/labkey/experiment/controllers/exp/ExperimentUpdateView.java -text
 experiment/src/org/labkey/experiment/controllers/exp/exportSampleSetAsXar.jsp -text
+experiment/src/org/labkey/experiment/controllers/exp/experimentRunGraphView.jsp -text
```
I don't think we need to add new files to the .gitattributes list -- only for the legacy files moved from svn.  @labkey-klum or @labkey-tchad may know.

labkey-adam replied:
That's my understanding as well

labkey-nicka replied:
Fixed. Thanks for the info.

labkey-nicka committed:
Bump @labkey/components

labkey-nicka committed:
Bump @labkey/components

labkey-nicka committed:
Code review feedback

labkey-nicka committed:
Code review feedback

labkey-adam committed:
Earliest upgrade version is now 18.10. Consolidate bootstrap scripts to 0.00-18.10.sql. (#1015)

labkey-kevink committed:
small fixes to LABKEY.Experiment.loadRun (#1022)

- always include material and data input/outputs for runs without protocol applications
- only include 'role' for materials until Issue 40119 is addressed

Fixes ModuleAssayTranformTest, FormulationsTest, SignalDataRawTest

labkey-klum committed:
allow .specimens files for the specimen file watcher (#1020)

labkey-nicka committed:
Bump @labkey/components

labkey-nicka committed:
Bump @labkey/components

cnathe committed:
Remove unused code from the BaseSampleSetAction

* Remove unused code from the BaseSampleSetAction now that it has been converted to new sample type designer workflow
* Add back in call to clearMaterialSourceCache during sample set update workflow

labkey-nicka committed:
Enable @labkey/api via experimental flag (#1021)

labkey-tchad committed:
Remove tests from 'FileBrowser' suite (#1027)

These tests interact with the file browser but don't provide interesting coverage of it

labkey-nicka committed:
Add icon for runs in lineage

labkey-nicka committed:
Add icon for runs in lineage

labkey-susanh committed:
Item 6965: add searching for data classes (source types) (#1013)

labkey-nicka committed:
Bump @labkey/components

labkey-nicka committed:
Bump @labkey/components
